### PR TITLE
Support for DNS Provider: nic.ru

### PR DESCRIPTION
**Pull request by nordicdyno**
Add support for the most popular DNS in Russia)
It tested manually and works fine in our case

### CODE REVIEW

File: providers/dns/nicru/nicru.go [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+	provider := &DNSProvider{
+		baseURL: baseURL,
+
+		apiUsername: nicruUsername,
+		apiPassword: nicruPassword,
+		apiID:       nicruAPIKey,
+		apiSecret:   nicruAPISecret,
+		serviceName: nicruServiceName,
+
+		client: http.Client{Timeout: 30 * time.Second},
```
Comment (mholt) [inline providers/dns/nicru/nicru.go]: Am thinking this should use `acme.HTTPClient` instead

File: providers/dns/nicru/nicru.go [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+	if err != nil {
+		return nil, err
+	}
+	return provider, nil
+}
+
+// getCredentials uses the supplied credentials to return a
+// DNSProvider instance configured for nic.ru
+func (d *DNSProvider) getAuthRespone() error {
+	// TODO: validate all credential are set
```
Comment (mholt) [inline providers/dns/nicru/nicru.go]: Should we wait to merge this until this is done? I imagine if credentials aren't set, the requests fail anyway.

File: providers/dns/nicru/nicru.go [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return fmt.Errorf("nicru auth failed, code=%v with unreadable body (%s)", resp.StatusCode, err)
+		}
+
+		var errInfo authError
+		err = json.Unmarshal([]byte(body), &errInfo)
+		if err != nil {
+			return fmt.Errorf("nicru auth failed, code=%v without parsable error: %s, body: %s", resp.StatusCode, err, body)
+		}
```
Comment (mholt) [inline providers/dns/nicru/nicru.go]: You can simplify this whole section by using `json.NewDecoder(resp.Body).Decode(&errInfo)` like you do below

File: providers/dns/nicru/nicru.go [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", d.auth.AccessToken))
+
+	resp, err := d.client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		body, err := ioutil.ReadAll(resp.Body)
```
Comment (mholt) [inline providers/dns/nicru/nicru.go]: This could be moved outside the `if` statement because you do the same thing below too.

**Comment by nordicdyno** [comment]
@ldez sorry, I don't have access to nic.ru account on my current work, and also no need this solution anymore.

**Comment by ldez** [comment]
I close this PR because it can not be finalized or maintained by the contributor.

Thanks anyway for the contribution.

**Pull request closed without merging**